src/java.base/share/classes/java/security/Security.java

```diff
@@ -141,8 +141,12 @@ private static void initialize() {
         }
 
         /*[IF CRIU_SUPPORT]*/
-        // Check if CRIU checkpoint mode is enabled, if it is then reconfigure the security providers.
-        if (InternalCRIUSupport.isCheckpointAllowed()) {
+        // Check if CRIU checkpoint mode is enabled.
+        // Check if the -XX:-CRIUSecProvider JVM option is enabled, if it is not, then use the CRIU security provider.
+        // Otherwise, use the normal providers from the java.security file.
+        if (InternalCRIUSupport.isCheckpointAllowed()
+            && !InternalCRIUSupport.disableCRIUSecProvider()
```

Comment:
Based on the discussion in https://github.com/eclipse-openj9/openj9/issues/18422 (see https://github.com/eclipse-openj9/openj9/issues/18422#issuecomment-1812483952), I think we also want `NativeCrypto.getVersionIfAvailable()` to check `disableCRIUSecProvider()`.

Reply:
A consequence of my more recent suggestion in https://github.com/eclipse-openj9/openj9/pull/18354#pullrequestreview-1739930746, is that this can be simplified to
```java
        if (InternalCRIUSupport.enableCRIUSecProvider()) {
```
which makes the comments on lines 144-146 of questionable value.

Reply:
Updated